### PR TITLE
Component layouts

### PR DESCRIPTION
**Pull request by elmerzsolt**

### CODE REVIEW

File: css/components.css [hunk 1]
```diff
@@ -1,73 +1,218 @@
+.background{
+  background-color: #f4f3f5;
+}
+
+
+
+
+/*TODO: A header elemei jelenleg nem mennek ki eléggé szélre, ezt még meg kell oldani.*/
+
 .header {
```
Comment (laszlovad) [inline css/components.css:10]: The header on my computer is completely broken:

Screen Shot 2020-04-02 at 10 58 17

I suggest the following solution for the header:
- Conceptually the header is a full-width container;
- with a 70% wide nested container inside it.
- This nested container holds three boxes:
  - one that contains the subscribe button;
  - one that contains the logo;
  - and finally one that holds the search and signup buttons.

The first two "boxes" can be the elements themselves, and I'd add a div that holds the search and signup buttons. I'd float all three boxes to the left and add the appropriate right margin for the first two. This would push all elements into its proper position.

Be aware, you also need to make the search and signup buttons block elements (anchors are inline by default), then float them to be able to properly position them beside each other.

File: index.html [hunk 2]
```diff
@@ -11,238 +11,276 @@
...
+        </div>
 
       </div>
 
 
-      <div class="gutter--horizontal bordered blogpost">
+      <div class="layout__item--33-percent-w-gutter layout__item--floated-right">
 
-        <h6>January 1, 2020 / by Chris / 7 Comments</h6>
+        <div class="gutter--horizontal">
```
Comment (laszlovad) [inline index.html:153]: This element should be the actual `.sidebar-card`. The distinct components should hold everything in them, title, lists and the button at the bottom.

Also, while I know that the UI kit points to this component as the "sidebar card", I think that's inaccurate and misleading naming. This is a `.sidebar-container` that can hold a lot of stuff.

Yes, the design is card-based but that's A) a visual thing and we should name stuff functionally, B) based on this logic we should build everything from a single card component (blog posts, category cards, sidebar). That would be doable but I think, ill-advised because these elements have nothing in common apart from their appearance.

Long story short, please rename the component. :)

File: index.html [hunk 3]
```diff
@@ -11,238 +11,276 @@
...
-            <h6>HEALTH</h6>
+              <p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</p>
 
-            <p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</p>
+            </div>
 
           </div>
 
-          <div>
+          <div class="sidebar-card__bottom-link">
```
Comment (laszlovad) [inline index.html:195]: I'd name these elements `sidebar-container__call-to-action` because that's their purpose.

File: css/components.css [hunk 4]
```diff
@@ -1,73 +1,218 @@
...
+      height:         22.857rem;
+      width:          100%;
+      border-radius:  1.071rem 1.071rem 0 0;
+    }
+
+
+
+
+
+.category-card {
```
Comment (laszlovad) [inline css/components.css:145]: This component is beautifully done. :)

File: css/components.css [hunk 5]
```diff
@@ -1,73 +1,218 @@
...
   /*opacity: 0.3;*/
+  padding:          1.429rem;
+  background-color: #ffffff;
 }
 
+  .category-card--rounded {
```
Comment (laszlovad) [inline css/components.css:151]: I'd add the border-radius to the base component, and create a class called `.category-card--nested` where I remove the border radius. The base component in my opinion is the standalone card, and sometimes these cards are nested into sidebars, which is a variant.

File: css/components.css [hunk 6]
```diff
@@ -1,73 +1,218 @@
...
+  .blogpost__imaged-blogpost {
+    border-radius: 0 0 1.071rem 1.071rem;
+  }
+
+/* TODO: A kép ott van a poszt fölött, de most nem kitölti a helyet, csak széthúzódik. */
+
+    .blogpost__imaged-blogpost-image {
+      height:         22.857rem;
+      width:          100%;
+      border-radius:  1.071rem 1.071rem 0 0;
```
[the remark this answers is not in the document]
Comment (elmerzsolt) [reply css/components.css:138]: But if I leave it in px here, and use rem in the button section, than it's not really consistent. If it's okay for you, I'd rather leave it in rem (not understandable and readable less then in other places in the code I think), but I'm persuadable. :)

File: index.html [hunk 7]
```diff
@@ -11,238 +11,270 @@
...
 
-      <a class="subscribe">SUBSCRIBE</a>
-      <a href="index.html" class="header__logo">LOGO TEXT</a> <a class="search"><i class="fa fa-search"></i></a>
-      <a class="sign-up">SIGN UP</a>
+<!--        <div class="layout__item&#45;&#45;70-percent">-->
 
-    </header>
+          <a class="button button--small button--filled-and-white text--uppercase header__subscribe">Subscribe</a>
+          <a href="index.html" class="header__logo"><img src="assets/logo_50.svg" alt=""></a>
+          <a class="button button--small button--transparent-and-white text--uppercase header__search">
```
[the remark this answers is not in the document]
Comment (elmerzsolt) [reply index.html:30]: But somehow I have to set a margin between the search and the sign up buttons, how do I do that if I can't declare it in a class?

File: css/components.css (hunk 6 above)
[the remark this answers is not in the document]
Comment (laszlovad) [reply css/components.css:125]: It's perfectly okay with me. This is a flavor question at this point.

File: index.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (laszlovad) [reply index.html]: Yeah, I ended up needing a class as well, so you were right. :) Thanks for pushing back on this. I try to understand your code as deeply as possible during the review but it's always possible that I don't get the full necessity of a particular solution. Please keep your good habit and challenge me whenever you don't agree. I might be wrong. :)
